rpcserverhandlers_test.go

```diff
@@ -6,28 +6,821 @@ package main
+	for i := range hashes {
+		switch hashes[i].String() {
+		case "1189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39",
+			"2189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39":
+			existsSlice[i] = true
+		case "3189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39",
+			"4189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39":
+			existsSlice[i] = false
+		case "5189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39":
+			return []bool{false, false}
+		}
+	}
+	return existsSlice
+}
+
+// CheckLiveTicket returns a mocked result of returns whether or not a ticket
+// exists in the live ticket treap of the best node.
+func (c *testRPCChain) CheckLiveTicket(hash chainhash.Hash) bool {
+	switch hash.String() {
+	case "1189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39":
+		return true
+	default:
+		return false
+	}
+}
+
+// CheckLiveTickets returns a mocked slice of bools representing
+// whether each ticket exists in the live ticket treap of the best node.
+func (c *testRPCChain) CheckLiveTickets(hashes []chainhash.Hash) []bool {
+	existsSlice := make([]bool, len(hashes))
+	for i := range hashes {
+		switch hashes[i].String() {
+		case "1189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39",
+			"2189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39":
+			existsSlice[i] = true
+		case "3189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39",
+			"4189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39":
+			existsSlice[i] = false
+		case "5189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39":
+			return []bool{false, false}
+		}
+	}
+	return existsSlice
+}
+
+// CheckMissedTickets returns a mocked slice of bools representing
+// whether each ticket hash has been missed
+func (c *testRPCChain) CheckMissedTickets(hashes []chainhash.Hash) []bool {
+	existsSlice := make([]bool, len(hashes))
+	for i := range hashes {
+		switch hashes[i].String() {
+		case "1189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39",
+			"2189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39":
+			existsSlice[i] = true
+		case "3189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39",
+			"4189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39":
+			existsSlice[i] = false
+		case "5189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39":
+			return []bool{false, false}
+		}
+	}
+	return existsSlice
+}
+
+// ConvertUtxosToMinimalOutputs returns a mocked MinimalOutput slice based on the testTxHash
+func (c *testRPCChain) ConvertUtxosToMinimalOutputs(entry rpcserver.UtxoEntry) []*stake.MinimalOutput {
+	switch c.testTxHash.String() {
+	case "3189cbe656c2ef1e0fcb91f107624d9aa8f0db7b28e6a86f694a4cf49abc5e39":
+		return []*stake.MinimalOutput{{
+			PkScript: []byte{0xBA, 0xA9, 0x14, 0x78, 0x02, 0x39, 0xEA, 0x12, 0x31, 0xBA, 0x67, 0xB0, 0xC5, 0xB8, 0x2E, 0x78, 0x6B, 0x51, 0xE2, 0x10, 0x72, 0x52, 0x21, 0x87},
```

Comment:
Could you break up these long lines? Here and in other places throughout the tests.

Reply:
Sure, I will adjust that. Thanks for your comments!